.github/workflows/lint.yml

```diff
@@ -16,14 +16,22 @@ jobs:
         python-version: ${{ matrix.python-version }}
     - name: Install dependencies
       run: |
+        python -m venv ~/.venv/ci-venv
+        source ~/.venv/ci-venv/bin/activate
         python -m pip install --upgrade pip
         pip install -r **/requirements.txt
         pip install -r **/requirements-dev.txt
-        echo "Final dependency list:"
-        pip freeze
     - name: Linting with ruff
       run: |
+        source ~/.venv/ci-venv/bin/activate
```

Comment:
having to reactivate the venv on each step is not so nice, lmk if there's a better way to do this with github actions?

Reply:
Hm, I'm surprised Maturin needs venv (I run without venv locally; use conda). Let me check